src/views/Gantt.vue

```diff
@@ -0,0 +1,220 @@
+  'platform'
+]
+/** The query which retrieves historical Job timing statistics */
+const QUERY = gql`
+query ganttQuery ($workflows: [ID]) {
+  jobs(live: false, workflows: $workflows) {
+    ${jobFields.join('\n')}
+  }
+}
+`
+/** The callback which gets called when data comes in from the query */
+class GanttCallback {
+  /**
+   * @param {Object[]} jobs
+   */
+  constructor (jobs) {
+    this.jobs = jobs
+  }
+  /**
+   * Add jobs contained in data to this.jobs
+   */
+
+  add (data) {
+    this.jobs.push(
+      ...data.jobs.map((job) => pick(job, jobFields))
+    )
+  }
+  // called when new objects are added
+  // NOTE: we manually call this to add items which come through on the query
+
+  onAdded (added, store, errors) {
+    this.add(added)
+  }
+
+  // called when existing objects are updated
+  onUpdated (updated, store, errors) {
+    this.add(updated)
+  }
+
+  // other hooks we don't need but must declare (for now)
+  before () {}
+  after () {}
+  onPruned () {}
+  commit () {}
+  tearDown () {}
+}
+export default {
+  name: 'Gantt',
+  mixins: [
+    graphqlMixin
+  ],
+  components: {
+    GanttChart,
+  },
+  head () {
+    return {
+      title: getPageTitle('App.workflow', { name: this.workflowName })
+    }
+  },
+  beforeMount () {
+    this.jobsQuery()
+  },
+  data () {
+    const jobs = []
+    return {
+      currentPage: 1,
+      callback: new GanttCallback(jobs),
+      /** Object containing all of the jobs added by the callback */
+      jobs,
+      timingOptions: [
+        { value: 'totalTimes', title: 'Total times' },
+        { value: 'runTimes', title: 'Run times' },
+        { value: 'queueTimes', title: 'Queue times' },
+      ],
+      jobsFilter: {
+        name: '',
+        timingOption: 'totalTimes',
+        platformOption: -1,
+      },
+    }
+  },
+  computed: {
+    // a list of the workflow IDs this view is "viewing"
+    // NOTE: we plan multi-workflow functionality so we are writing views
+    // to be mult-workflow compatible in advance of this feature arriving
+    workflowIDs () {
+      return [this.workflowId]
+    },
+    filteredJobs () {
+      return this.jobs.filter(job => matchTask(job, this.jobsFilter))
+    },
+    platformOptions () {
+      return platformOptions(this.jobs)
+    },
+    timingOption () {
+      return this.jobsFilter.timingOption.replace(/Times/, '')
+    },
+  },
+  methods: {
+    /**
+     * Run the one-off query for historical job data and pass its results
+     * through the callback
+     */
+    jobsQuery: debounce(
+      async function () {
+        this.jobs = []
+        this.callback = new GanttCallback(this.jobs)
+        const ret = await this.$workflowService.query2(
+          QUERY,
+          { workflows: this.workflowIDs }
+        )
+        this.callback.onAdded(ret.data)
+      },
+      200 // only re-run this once every 0.2 seconds
+    ),
+    onPageChange (page) {
+      console.log(page)
+      this.currentPage = page
+    }
```

Comment:
Unused?

Reply:
Previously a pagination was going to be included. But some complications have caused my to remove it and add it in as a pull request later. Missed a few deps